template/src/pageComponents/buy-me-coffee/steps/BuyCoffeeFormStep/FormBuyCoffee.tsx

```diff
@@ -43,64 +43,90 @@ function FormBuyCoffee({
   const canAfford = useLoggedInUserCanAfford(parseEther(String(buyCoffeeAmount)));
 
   // Wagmi Write call
-  const { config } = usePrepareContractWrite({
+  const { data } = useSimulateContract({
     address: contract.status === 'ready' ? contract.address : undefined,
     abi: contract.abi,
     functionName: 'buyCoffee',
     args: [BigInt(numCoffees), name, twitterHandle, message],
-    enabled: name !== '' && message !== '' && contract.status === 'ready',
     value: parseEther(String(buyCoffeeAmount)),
-    onSuccess(data) {
-      console.log('Success prepare buyCoffee', data);
+    // onSuccess(data) {
+    //   console.log('Success prepare buyCoffee', data);
+    // },
+    query: {
+      enabled: name !== '' && message !== '' && contract.status === 'ready',
     },
   });
 
   // Wagmi Write call
-  const { write: buyMeACoffee, data: dataBuyMeACoffee } = useContractWrite({
-    ...config,
-    onSuccess(data) {
-      console.log('Success write buyCoffee', data);
+  const {
+    writeContract: buyMeACoffee,
+    data: dataBuyMeACoffee,
+    status: statusBuyMeACoffee,
+    error: errorBuyMeACoffee,
+  } = useWriteContract();
+
+  const { status: transactionStatus } = useWaitForTransactionReceipt({
+    hash: dataBuyMeACoffee,
+    query: {
+      enabled: !!dataBuyMeACoffee,
+    },
+    // onSuccess() {
```

Comment:
Do we need to handle these?

Reply:
onSuccess and onError callbacks have been removed in wagmi@2